### PR TITLE
Refs [TASK][26256] Init Database

### PR DESCRIPTION
**Pull request by quockhanhng**
## Related Tickets
- [#26256](https://edu-redmine.sun-asterisk.vn/issues/26256)

## WHAT
- Init Database

## HOW
- I create AppDatabase class implement SQLiteOpenHelper class and implement required functions

## WHY
- To store local data in the app

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Reviewer1 (KhanhNQ) | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/getitdone/data/source/local/database/AppDatabase.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import android.database.sqlite.SQLiteOpenHelper
+import com.sunasterisk.getitdone.utils.Database
+
+class AppDatabase private constructor(context: Context, dbName: String, version: Int) :
+    SQLiteOpenHelper(context, dbName, null, version) {
+
+    override fun onCreate(db: SQLiteDatabase?) {
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/getitdone/data/source/local/database/AppDatabase.kt]: xóa bỏ cái ? kia thì đỡ phải check null có được k ô nhỉ ?

File: app/src/main/java/com/sunasterisk/getitdone/utils/Constants.kt [hunk 2]
```diff
@@ -14,3 +14,28 @@ object Constants {
...
+
+object Database {
+    const val DATABASE_NAME = "GET_IT_DONE_DATABASE"
+    const val DATABASE_VERSION = 1
+
+    const val SQL_CREATE_TABLE_LIST = "CREATE TABLE LISTS (" +
+            "ID INTEGER PRIMARY KEY AUTOINCREMENT, " +
+            "TITLE TEXT)"
+
+    const val SQL_CREATE_TABLE_TASK = "CREATE TABLE TASKS (" +
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/getitdone/utils/Constants.kt]: t nhớ không lầm là tên table đâu cần phải có thêm " S" như v nhỉ. USER chứ t k thấy a đặt là USERS cả

File: app/src/main/java/com/sunasterisk/getitdone/data/source/local/database/AppDatabase.kt (hunk 1 above)
Comment (quockhanhng) [reply app/src/main/java/com/sunasterisk/getitdone/data/source/local/database/AppDatabase.kt]: được ạ :sweat_smile: tại default khi nó tự implement là nullable nên em vẫn để thế

File: app/src/main/java/com/sunasterisk/getitdone/utils/Constants.kt (hunk 2 above)
Comment (quockhanhng) [reply app/src/main/java/com/sunasterisk/getitdone/utils/Constants.kt]: fixed :bow:

**Pull request merged**